prompt2model/utils/config.py

```diff
@@ -0,0 +1,31 @@
+# This is something we need to decide, what should be the max value
+# because the more the max value, the more it will take time to get the
+# max batch size
+
+MAX_SUPPORTED_BATCH_SIZE = 128
```

Comment:
This is probably too big. I think something like 12 or 16 makes more sense (since we're assuming that folks will be training BERT-sized models on small GPUs).

Reply:
I am also adding a function that I will put on the next commit, to automatically fix the max batch size. This goes something like this

```python
# this is something we need to know
MAX_SUPPORTED_BATCH_SIZE = 128


def get_max_batch_size(model: nn.Module, input_dim: int, device: str):
    batch_size = MAX_SUPPORTED_BATCH_SIZE

    while True:
        try:
            dummy_tensor = torch.rand(batch_size, input_dim).to(device)
            _ = model(dummy_tensor)
            break
        except RuntimeError as e:
            if "CUDA out of memory" in str(e):
                batch_size //= 2
            else:
                raise e
    return batch_size

```

Reply:
I think we can add this in a follow-up PR? For now we can just use a small batch size by default for simplicity.

Reply:
Yes, we can do it